### PR TITLE
update diligent-core package layout

### PR DESCRIPTION
**Pull request by AndreyMlashkin**
Specify library name and version:  **diligent-core/2.5.1**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by AndreyMlashkin** [comment]
This fixes RelWithDebInfo build.
Interesting, that there is no build in CI with this build_type

**Comment by AndreyMlashkin** [comment]
@SSE4 @uilianries @jgsogo @SpaceIm 
please, review

I haven't found a way to use self.move or something like this

**Comment by uilianries** [comment]
Do you have a log which exemplifies your error?



> This fixes RelWithDebInfo build. Interesting, that there is no build in CI with this build_type

We only build Release and Debug: https://github.com/conan-io/conan-center-index/blob/master/docs/supported_platforms_and_configurations.md#windows
One more configuration requires more storage, more CI time, more maintenance, but few usage ... too expensive

> I haven't found a way to use self.move or something like this

Indeed there is no such tool. The closer helper is `tools.rename` which is a safer way for `os.rename` on Windows.

**Comment by AndreyMlashkin** [comment]
The error was in a proprietary code. It wasn't really an error, but the package would be invalid, as self.cpp_info.libdirs would not point to the libs location.

and I think, it's better to have the libs in a standard location under lib/ and bin/

**Comment by AndreyMlashkin** [comment]
I tried to use tools.rename, but it only have effect, if you rename the last folder. It does not work with the paths.

**Comment by uilianries** [comment]
@AndreyMlashkin Did you accept the CLA? The validation is still waiting. Sometimes is a bug and we need to re-run.

Screenshot 2022-02-22 at 11-15-12 update diligent-core package layout by AndreyMlashkin · Pull Request #9451 · conan-io con

**Comment by AndreyMlashkin** [comment]
Yes, I did. Now I rejected and accepted again

**Review by uilianries — APPROVED** [review]
LGTM

**Review by SSE4 — APPROVED** [review]

**Comment by AndreyMlashkin** [comment]
@jgsogo @SpaceIm @prince-chrismc @ericLemanissier 
please, review

**Review by ericLemanissier — DISMISSED** [review]

File: recipes/diligent-core/all/conanfile.py [hunk 1]
```diff
@@ -146,12 +148,21 @@ def package(self):
...
         self.copy("License.txt", dst="licenses", src=self._source_subfolder)
 
-    def package_info(self):
-        if self.settings.build_type == "Debug":
-            self.cpp_info.libdirs.append("lib/source_subfolder/Debug")
-        if self.settings.build_type == "Release":
-            self.cpp_info.libdirs.append("lib/source_subfolder/Release")
+        bin_path = os.path.join(self.package_folder, "lib", "source_subfolder", str(self.settings.build_type))
+
+        for dll in glob.glob(os.path.join(self.package_folder, bin_path, "*.dll")):
```
Comment (ericLemanissier) [inline recipes/diligent-core/all/conanfile.py:153]: Now that you're in package(self), you can use self.copy, and remove the glob and shutil import

**Review by ericLemanissier — APPROVED** [review]
This `cmake.install()` call seems to be close to useless

**Comment by AndreyMlashkin** [comment]
Agree...

**Review by jgsogo — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Comment by AndreyMlashkin** [comment]
I don't know, where to ask this question, so I will try here:

I need to package an internal tool 'File2String' for diligent-core.
For this I have added locally to a package method:

        self.copy(os.path.join(self.build_folder, "build_subfolder", "bin", "File2String"), dst="bin", keep_path=False)
        print(os.path.join(self.build_folder, "build_subfolder", "bin", "File2String"))

It's not packaged after this command. I have double checked if the file exists by enetering a path, which was printed. The file is there

**Comment by SSE4** [comment]
> I don't know, where to ask this question, so I will try here:
> 
> I need to package an internal tool 'File2String' for diligent-core. For this I have added locally to a package method:
> 
> ```
>     self.copy(os.path.join(self.build_folder, "build_subfolder", "bin", "File2String"), dst="bin", keep_path=False)
>     print(os.path.join(self.build_folder, "build_subfolder", "bin", "File2String"))
> ```
> 
> It's not packaged after this command. I have double checked if the file exists by enetering a path, which was printed. The file is there

I don't remember all the rules fully, but as I understand `self.copy('myfile')` will copy nothing, while it has to be `self.copy('*myfile')` (with asterisk), that's how matcher works

**Comment by AndreyMlashkin** [comment]
I have found a solution and started a PR, thanks! :)

**Pull request merged**